prover/src/lib.rs

```diff
@@ -33,14 +36,15 @@ pub use winter_prover::StarkProof;
 ///
 /// # Errors
 /// Returns an error if program execution or STARK proof generation fails for any reason.
-pub fn prove<A>(
+pub fn prove<A, H>(
     program: &Program,
     stack_inputs: StackInputs,
     advice_provider: A,
-    options: &ProofOptions,
-) -> Result<(StackOutputs, StarkProof), ExecutionError>
+    options: &ProofOptions<H>,
+) -> Result<MidenProof, ExecutionError>
```

Comment:
Let's not add `H` as the generic parameter here. Ideally, the info about which hash function to use should come from `ProofOptions` via an enum.

However, we may need to wrap `StarkProof` into something which also contains information about the hash function used.